### PR TITLE
feat(developer): Support v2.0 projects in TIKE 🦕

### PR DESCRIPTION
**Pull request by mcdurdin**
Relates to #9948. See that issue for remaining work items. 

This PR starts basic support for v2.0 projects.

For all operating modes:

- Remove blank projects / untitled projects mode
- Add Open Project Folder buttons and links
- Remove Project Save As
- Render uses in-memory XML
- Upgrade to v2.0 project file (along with checks)

For v2.0 projects:

- Remove 'Add File to Project'
- Populate Files to list all files in folder (note: consider efficiency
  here as currently files are parsed for subfile data)
- Version 2.0 project options and defaults

# User Testing

**TEST_OPEN_PROJECT:** Verify that opening an existing project continues to function correctly. Try basic project functions, such as opening files in the project, building keyboards (from the buttons in the Project view), adding files.

**TEST_NEW_PROJECT:** Verify that creating a new basic keyboard project continues to function correctly.

**TEST_UPGRADE_PROJECT:** When opening an existing project, an Upgrade banner should appear. Click the button to upgrade the project, and follow the prompts. Load the result .kpj file in a text editor and verify (a) that the `<Version>` field now reads '2.0', and (b) that there are no `<File>` entries in the .kpj file.

### CODE REVIEW

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_OPEN_PROJECT (PASSED):** Tested with the attached PR build (Keyman Developer 17.0.204-alpha-test-9949) on Windows 10 OS and here is my observation: 1. Verified that opening an existing project continues to function correctly. Tried basic project functions such as opening files in the project and building keyboards in a project and it seems to be working as expected. 2. After adding a file from another project (eg., history.md or another .kpj file)., I got a warning message in the Project - Keyboards screen. 3. Clicking on the Upgrade Project button shows another error message on the screen. Is this an expected one? other than this, it seems to be working as expected. 

..warning message 1

**Comment by mcdurdin** [comment]
> * Is this an expected one?

Yes, that is an expected message, looks good thank you. (A future PR will also make those messages clearer)

**Comment by bharanidharanj** [comment]
* **TEST_NEW_PROJECT (PASSED):** Tested with the attached PR build (Keyman Developer 17.0.204-alpha-test-9949) on Windows 10 OS and here is my observation: 1. Created a Tamil Keyboard project. Added Tamil keyboard layout in the Keyboard Editor. Compiled it. 2. Open the keyboard in the localhost browser using the 'Test keyboard on web' button. 3. Able to see the 'Tamil fonts' in some device tools (like Windows, macOS, and Linux OS). 4. But, it's not working while opening in iPhone / Pixel 5 devices. Seems to be an issue. Or Is this an expected one? Besides this, I don't see any other issues. 

..Tamil keyboard in Windows 

   

..Tamil keybaord in iPhone

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_UPGRADE_PROJECT (PASSED):** Tested with the attached PR build (Keyman Developer 17.0.204-alpha-test-9949) on Windows 10 OS and here is my observation: 1. Clicked the 'Upgrade project button to upgrade the project. 2. Opened the .kpj file in a text editor. 3. Verified that the file includes <Version> 2.0  number and there are no <File> entries in the .kpj file. Seems to be working as expected.

File: developer/src/tike/main/UfrmMain.pas [hunk 1]
```diff
@@ -1159,8 +1157,16 @@ function TfrmKeymanDeveloper.OpenFile(FFileName: string; FCloseNewFile: Boolean)
...
     // This can happen if we get a file opened via Explorer.
-    modActionsMain.NewProject(ptKeyboard);
+    ShowMessage('TODO -- open parent folder as project');
+    Exit(nil);
+  end;
+
+  if DirectoryExists(FFileName) then
+  begin
+    // This is an attempt to open a project folder?
+    // TODO
```
Comment (darcywong00) [inline developer/src/tike/main/UfrmMain.pas:1169]: Does this get a warning message?

Comment (mcdurdin) [reply developer/src/tike/main/UfrmMain.pas:1169]: No, it's a TODO to finish the functionality; captured in #9948.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.211-alpha](https://keyman.com/downloads/releases/alpha/17.0.211)

**Pull request merged**